test/cli/autocorrect-nilable-untyped/autocorrect-nilable-untyped.sh

```diff
@@ -0,0 +1,7 @@
+tmp="$(mktemp)"
+pre="test/cli/autocorrect-nilable-untyped/pre.rb"
+post="test/cli/autocorrect-nilable-untyped/post.rb"
+cp "$pre" "$tmp"
+main/sorbet --silence-dev-message -a "$tmp"
+diff "$post" "$tmp"
```

Comment:
the slightly annoying thing about structuring the CLI test this way is that if this test ever needs to be updated, the bazel `update` test target will write the (new, non-empty) diff file into the `.out` file, instead of changing `post.rb`.

If you structure your autocorrect test like this, it will play better with the update scripts, and still show good `diff` output when the autocorrect output is not what was expected:

https://github.com/sorbet/sorbet/pull/4786/files#diff-0dabd4297161e4f7ab7e711a40cdb181021df3f46a206de93e6f930d1a6733f5R21

Reply:
my guess is that you're probably just cargo culting something here. if you want to change in this PR, go ahead. otherwise, i'll go ahead and change it after you land, and also change all the other tests that use this pattern.

Reply:
Yep, just cargo-culted existing autocorrect cli tests. I'm happy to make the output the diff instead.